SU2_CFD/src/solvers/CIncEulerSolver.cpp

```diff
@@ -1490,6 +1495,80 @@ void CIncEulerSolver::Source_Residual(CGeometry *geometry, CSolver **solver_cont
 
   }
 
+  if (streamwise_periodic) {
+    numerics->SetStreamwise_Periodic_Values(Streamwise_Periodic_MassFlow, Streamwise_Periodic_IntegratedHeatFlow,
+                                            Streamwise_Periodic_InletTemperature);
+
+    /*--- Loop over all points ---*/
+    SU2_OMP_FOR_STAT(omp_chunk_size)
+    for (iPoint = 0; iPoint < nPointDomain; iPoint++) {
+
+      /*--- Load the primitive variables ---*/
+      numerics->SetPrimitive(nodes->GetPrimitive(iPoint), nullptr);
+
+      /*--- Set incompressible density ---*/
+      numerics->SetDensity(nodes->GetDensity(iPoint), 0.0);
+
+      /*--- Load the volume of the dual mesh cell ---*/
+      numerics->SetVolume(geometry->nodes->GetVolume(iPoint));
+
+      /*--- If viscous, we need gradients for extra terms. ---*/
+      if (viscous) {
+        /*--- Gradient of the primitive variables ---*/
+        numerics->SetPrimVarGradient(nodes->GetGradient_Primitive(iPoint), nullptr);
+      }
+
+      /*--- Compute the streamwise periodic source residual and add to the total ---*/
+      auto residual = numerics->ComputeResidual(config);
+      LinSysRes.AddBlock(iPoint, residual);
+
+      /*--- Add the implicit Jacobian contribution ---*/
+      if (implicit) Jacobian.AddBlock2Diag(iPoint, residual.jacobian_i);
+
+    }// for iPoint
+
+    if(!streamwise_periodic_temperature && energy) {
+      CNumerics* second_numerics = numerics_container[SOURCE_SECOND_TERM];
+      second_numerics->SetStreamwise_Periodic_Values(Streamwise_Periodic_MassFlow, Streamwise_Periodic_IntegratedHeatFlow,
+                                                     Streamwise_Periodic_InletTemperature);
+
+      /*--- This bit acts as a boundary condition rather than a source term. But logically it fits better here. ---*/
+      for (auto iMarker = 0ul; iMarker < config->GetnMarker_All(); iMarker++) {
+
+        /*--- Only "inlet"/donor periodic marker ---*/
+        if (config->GetMarker_All_KindBC(iMarker) == PERIODIC_BOUNDARY &&
+            config->GetMarker_All_PerBound(iMarker) == 1) {
+
+          for (auto iVertex = 0ul; iVertex < geometry->nVertex[iMarker]; iVertex++) {
```

Comment:
Then on top of this loop SU2_OMP_FOR_STAT(OMP_MIN_SIZE) (or something)

Reply:
Yes. following the other loops:
```c++
SU2_OMP_FOR_STAT(omp_chunk_size)
for (auto iVertex = 0ul; iVertex < geometry->nVertex[iMarker]; iVertex++) {
```

Reply:
omp_chunk_size is for nPoint loops, for ones over vertices should be OMP_MIN_SIZE which is a smaller value.
I think there is a "nVertex" in CSolver, getting it from geometry is not needed.